### ISSUE TITLE
harden-runner Github action step just hangs indefinitely

### ISSUE THREAD

**Issue by jaypipes** [step-security/harden-runner#466]
Hi!

I've been a fan of step-security/harden-runner for a while and used it successfully on my open source projects. However, recently, I noticed the step-security/harden-runner step just hanging indefinitely. I've [had to comment out](https://github.com/jaypipes/ghw/pull/377) the harden-runner step from my workflows in order to get our test and linter workflows working again.

Here is the YAML that consistently results in hung step execution:

```yaml
     - name: harden runner
       uses: step-security/harden-runner@55d479fb1c5bcad5a4f9099a5d9f37c8857b2845 # v2.4.1
       with:
         egress-policy: block
         disable-sudo: true
         allowed-endpoints: >
           github.com:443
           api.github.com:443
           proxy.github.com:443
           proxy.golang.org:443
           raw.githubusercontent.com:443
           objects.githubusercontent.com:443
           proxy.golang.org:443
```

Can you advise what the issue is with the above? Thanks much! :)

**Comment by jaypipes** [comment]
Here is an example of a stuck harden-runner step:

https://github.com/jaypipes/ghw/actions/runs/10961280262/job/30469312455

**Comment by varunsh-coder** [comment]
Hi @jaypipes thanks for reporting the issue! 

I noticed you were using a very old version of harden-runner (v2.4.1) and last year there was an issue that had caused this, which was fixed in v2.5.1. Can you please upgrade to the latest version v2.10.1?

**Comment by jaypipes** [comment]
@varunsh-coder yep, looks like that was all that was needed :) thank you sir!
